### PR TITLE
Expose service metadata and TSR

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?
- [x] Show TSR score on study card
- [x] Show detailed TSR scores and their references
- [x] Allow service owners edit TSR scores and their references
- [x] Allow service owners edit annotations
- [x] Add persistence

Notes:
- By default, all (and only) computational services get a TSR score (0/40). This could be enabled/disabled, let's see what POs say.

TSR
image

## Related issue number
related to https://github.com/ITISFoundation/osparc-issues/issues/356

## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: packages/models-library/src/models_library/services.py
Comment (pcrespov) [inline packages/models-library/src/models_library/services.py:289]: From the front-end I did a great step forward! On the other hand, the backend is totally blind to the metadata ... there is really a lot to do!

**Review by pcrespov — COMMENTED** [review]
I think you missed to update OAS in all services: follow the first step of the "check-list"

File: packages/models-library/src/models_library/services.py
Comment (GitHK) [reply packages/models-library/src/models_library/services.py:289]: I agree with @pcrespov here, the metadata field should at least have a validator for the metadata structure.

**Review by GitHK — COMMENTED** [review]
On my deployment there not all services have a metadata score (see blow image) and all others have a score of 0/40 (which was expected).

Screenshot 2020-12-14 at 08 53 58

When I open the metadata card I get this error
Screenshot 2020-12-14 at 08 54 40

Is there something I need to to in order to properly test it?

File: packages/models-library/src/models_library/services.py
Comment (pcrespov) [reply packages/models-library/src/models_library/services.py:289]: @GitHK We will have to take over that part in the next iteration. My point here is that we should not give the impression this is completed.

Comment (GitHK) [reply packages/models-library/src/models_library/services.py:289]: good to know!

**Comment by sanderegg** [comment]
@GitHK : it is ONLY for computational services

**Review by sanderegg — COMMENTED** [review]
Also get the issue that ```validation fails``` when clicking on Metadata in the dashboard on any of the computation services. I nailed it to the following, this.__copyMetadata is null:
```js
__populateAnnotationsData: function() {
      const copyMetadataAnnotations = this.__copyMetadata["metadata"]["annotations"];
```

Also when clicking on ```More Info``` in the workbench after selecting sleeper 0:
image

**Comment by KZzizzle** [comment]
SUPER ODEI!!! Could I just suggest a few changes to the labels below the TSR items:
Tooltip for "intended purpose/context": Description of the field of application and associated research initiatives (if any)
Tooltip for V&V: Description of verification and validation testing performed for the model and/or code submitted, or a link to these tests
Tooltip for documentation: Link to documentation to use this service
Tooltip for relevant standards: List or links of any data management, coding, experimental, or other standards implemented
Probably the other fields could use tooltips but since they seem yet to be defined well, there's no point to add them yet.

Suggestions: 

- In the service options, could this be named as "Quality Assessment" or something to that end vs "Metadata"? This is because all the other information that we already have like Authors, Description, inputs/outputs are also metadata.

- perhaps a "user-based rating" where something could be rated by people who don't have write access, such as other users? Sort of like a Netflix rating. It would have good "demo" potential, but I leave it to your judgement.

- I also leave to your judgement whether to implement is to have some sort of "unreviewed" status? For example no one has yet filled in any of these items and there should be some difference between actively reviewed and having 0/40 vs one that was not filled in at all.

File: packages/models-library/src/models_library/services.py
Comment (odeimaiz) [reply packages/models-library/src/models_library/services.py:289]: For now, the validation is only done in the frontend against this schema:
services/web/client/source/resource/form/service-metadata.json

**Comment by odeimaiz** [comment]
> * perhaps a "user-based rating" where something could be rated by people who don't have write access, such as other users? Sort of like a Netflix rating. It would have good "demo" potential, but I leave it to your judgement.
> *

Tooltips added and renamings done.

Regarding suggestions, in this first iteration, we only let service owners edit the quality assessment and I agree we should also let them decide (maybe in the second iteration) whether they want to enable/disable this evaluation.

**Review by sanderegg — APPROVED** [review]
very nicely done!

**Review by ignapas — COMMENTED** [review]
I couldn't throughout test this, how can I edit the data?

**Comment by odeimaiz** [comment]
> I couldn't throughout test this, how can I edit the data?

You need to be owner

File: services/web/client/source/class/osparc/component/metadata/ServiceQualityEditor.js
Comment (odeimaiz) [reply services/web/client/source/class/osparc/component/metadata/ServiceQualityEditor.js:52]: Yes, I can improve it in the next iteration

**Review by KZzizzle — COMMENTED** [review]
Go for it!

**Review by KZzizzle — APPROVED** [review]

**Pull request merged**
